### PR TITLE
Allow more multistage heating coils in AFN

### LINKED ISSUE #10389: Allow more multistage heating coils for AFN model
Issue overview
--------------
When a multiple speed heating pump is used, the heat pump allow 3 type of multispeed heating coils: Coil:Heating:DX:MultiSpeed, Coil:Heating:Electric:MultiStage, and Coil:Heating:Gas:MultiStage. However, the AFN model only allows Coil:Heating:DX:MultiSpeed. More multistage heating coil types should be allowed for users to select.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
 - Fixes #10389 
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

When multispeed heat pump is used in the AFN model, the allowed heating coil is DX heating coil and no multistage heating coils are allowed. The fix allows two more types heating coils in the AirflowNetwork:Distribution:Component:Coil:

      \key Coil:Heating:Electric:MultiStage
      \key Coil:Heating:Gas:MultiStage

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ X] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [X ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp [hunk 1]
```diff
@@ -10626,7 +10626,22 @@ namespace AirflowNetwork {
...
+                        SetHeatingCoilAirLoopNumber(m_state, DisSysCompCoilData(i).name, DisSysCompCoilData(i).AirLoopNum, ErrorsFound);
+                    }
+                } else if (SELECT_CASE_var == "COIL:HEATING:GAS:MULTISTAGE") {
+                    ValidateComponent(
+                        m_state, "Coil:Heating:Gas:MultiStage", DisSysCompCoilData(i).name, IsNotOK, format(RoutineName) + CurrentModuleObject);
+                    if (IsNotOK) {
+                        ErrorsFound = true;
+                    } else {
+                        SetHeatingCoilAirLoopNumber(m_state, DisSysCompCoilData(i).name, DisSysCompCoilData(i).AirLoopNum, ErrorsFound);
+                    }
```
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:10648]: Allow two more multistage heating coils

**Review by Myoldmopar — APPROVED** [review]
This seems extremely straightforward.  It does touch AFN, so I will ping @jasondegraw for comments, but it doesn't seem to really be doing anything controversial.  If anyone has thoughts, speak up!

**Comment by Myoldmopar** [comment]
I pulled develop in and resolved the unit test conflict.  It was just that two different unit tests were added to the bottom.  Everything builds and passes fine.  If no one has any additional comments, and this is fine on CI, I'll merge this today.

**Comment by Myoldmopar** [comment]
CI was all happy here, I'm merging this one in.  Thanks @lgu1234

**Pull request merged**